### ISSUE TITLE
Language files for translations

### ISSUE THREAD

**Issue by Ruvetuve** [ngovil21/PlexRequestChannel.bundle#16]
Nice Channel! Works like a charm. I want to request a language file option. This way i can translate it in my own language. Is this feasible?

**Comment by ngovil21** [comment]
Yes that is possible. I will start working on it. Lots of strings to go through.

**Comment by Ruvetuve** [comment]
Great! I remember for an other plex channel there is a website to start the translation of the strings in a ordered way. I'll try to find it.

**Comment by ngovil21** [comment]
I have added the localization to the development branch for now.

You will need to make a copy of the `en.json` file and name the file with the two letter code for your language. Translate the strings on the right side of the colon (:). If you see a %s, then the %s should stay in the same place in the translation.

If you would like me to add the translation file you make, you can submit a pull request, or just send it to me here.

Hope everything works.

**Comment by Ruvetuve** [comment]
Wow fast!  Let me check and translate

**Comment by seb972** [comment]
Hello,

I want to help this greatfull work with a French translation.

Can you say me if i can use accent as é à with the other (UTF-8 ?)

**Comment by seb972** [comment]
Hi,

Check my work to translate in french

[Uploading fr.zip…]()

**Comment by Ruvetuve** [comment]
Working on Dutch over here :)

**Comment by ngovil21** [comment]
@seb972 Thanks for the translation. I have added to the plugin, but in my testing it hasn't worked. Can you confirm on the latest branch if it works for you? 

May need to add some fixes for the way Plex handles languages. 

Do you know if language localization works in any other channel?

**Comment by seb972** [comment]
sorry i don't try languauge localization on other channel.
I try the development branch and i give you my feedback.

**Comment by ngovil21** [comment]
That's okay, most likely it won't work right now, but I have some fixes in mind.

**Comment by seb972** [comment]
how can i make a git clone on development branch?

**Comment by ngovil21** [comment]
Are you using the `git` tool, or are you downloading the .zip file?

If you are using git, go to the folder that has the plugin, type in `git checkout development` and then `git pull`.

If you are using the zip files, just wait until I merge into master. It'll be easier that way.

**Comment by seb972** [comment]
i use the git tool on linux, i try git clone -b development and it work, but the language not change.

**Comment by ngovil21** [comment]
OK that's what I thought. Hang on I'm going to have to figure this out.

**Comment by seb972** [comment]
Do you see this?

https://forums.plex.tv/discussion/95377/language-support-client-or-server-based

**Comment by ngovil21** [comment]
Yes, I was doing it correctly, but it's a problem with Plex. I found a fix that someone posted and it works. Please update the development branch and try it again.

**Comment by Ruvetuve** [comment]
Dunno how to submit Dutch, but here it is.

[nl.json.txt](https://github.com/ngovil21/PlexRequestChannel.bundle/files/166149/nl.json.txt)

**Comment by seb972** [comment]
I try and the French language is OK, the Channel is in my language (fr)

**Comment by pepicora** [comment]
I'm happy to help with an Italian Translation. 
I'll download the english file asap.

**Comment by pepicora** [comment]
Here it is the Italian translation, hope it will be ok with the UTF-8 unicode charset :)

[it.json.txt](https://github.com/ngovil21/PlexRequestChannel.bundle/files/300466/it.json.txt)

Thanks @ngovil21 for this project!!

**Comment by ngovil21** [comment]
@pepicora I have added the Italian translation to the master branch. Thank you for your help!

**Comment by pepicora** [comment]
@ngovil21 i've found some issues on the italian translation, i'll investigate better and i'll fix it, but can someone help me with the right utf code for this word: "Già"  i've used "GiÃ" but don't work.

Also on this line:

`"Please enter the name of the TV Show in the search box and press enter.": "Per favore inserisci il nome della Serie TV nel campo di ricerca e premi invio.",`

I translated right referring to TV Show but on the channel the popup ask for a Movie

Thanks

**Comment by ngovil21** [comment]
@pepicora Should all Gi be Già? I can make that change myself.

You're right about the "Please enter the TV Show" line. I have posted a fix.

**Comment by pepicora** [comment]
@ngovil21  here are the fixes that are needed under  the it.json file for the utf code

`già   output    gi   should be   già`

`GiÃ    output    gi?     should be   già`

`Ã©    output    é     should be   è`

`puÃ²   output     pu?   should be   può`

There are something not translated on the channel, here and there, but i can't find it under the en.json file, for example when you report a problem with a file the "missing subtitle" , "bad audio" ....
Also when a new user jump in for the first time it tell you "username registered" or something similar in english. Also the channel setup is in english. Is it possible to make a complete localization?

And finally i can't have the channel translated in italian under the android app, i've tryed all i can do but nothing.

Thank you so much for all your work on this channel!

**Comment by ngovil21** [comment]
@pepicora I think I have fixed the italian translation errors you pointed out. I didn't see any pu?, all  può show up correctly in my editor. Can you verify if its correct in Plex now?

I will add in the rest of the strings at another time, when I get around to it.

For Android, I don't see any way to change the language in the settings. Is the whole Plex app in Italian on Android? The client has to set the language, otherwise nothing I  an do with the channel.

**Comment by Tangoes** [comment]
Hi everybody!

I am translating to SPANISH, but I created es.json file, and nothing changes, the channel is still in English.

Any clue?

**Comment by ngovil21** [comment]
@Tangoes Could you attach, your es.json file here if you don't mind sharing? Plex has some bugs with how they handle translations, so I have implemented a work around. I can update the code to include your Spanish translation.

**Comment by Tangoes** [comment]
Hi @ngovil21 I didn´t translate it all yet. Just translated some strings, and it is not working.

Just copied en.json to en.json. If I translate en.json it works, so it´s a problem related to lang detection.

When I get the translate done, I´ll send it to you.

BR

**Comment by ngovil21** [comment]
Ok, to fix this, you will need to add 'es' to the LANGUAGES in __init___.py. If you are familiar with git you can make a fork and send me a pull request once you are done with the translation. Thank you for your contribution.
